src/library_syscall.js

```diff
@@ -40,9 +40,10 @@ var SyscallsLibrary = {
       }
       return PATH.join2(dir, path);
     },
-
-    doStat(func, path, buf) {
-      var stat = func(path);
+    // When called by stat, arg is a path. When called by fstat, arg is a file
+    // descriptor.
+    doStat(func, arg, buf) {
+      var stat = func(arg);
```

Comment:
Can we have the caller call `func(arg)` and just pass the stat result.

Maybe rename this `writeStat(bug, stat)`?

Reply:
Sounds good.